src/query/mpd_source.cpp

```diff
@@ -105,6 +105,7 @@ void mpd_source::refresh()
     struct mpd_connection* connection = connect();
     struct mpd_status* status = nullptr;
     struct mpd_song* mpd_song = nullptr;
+    m_current.clear();
```

Comment:
``m_current.clear()`` should ideally be after ``begin_refresh()``, as ``begin_refresh()`` saves the current song for comparison.

Reply:
Fixed.